src/widgets/PassageReferenceWidget.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+  <div class="passage-reference-widget u-widget">
+    <div class="healed" v-if="healed">
+      Passage reference <strong>{{ requested }}</strong> could not be resolved.
+      Instead, we are showing you <strong>{{ healed }}</strong>.
+    </div>
+    <div class="human" v-if="human">{{ human }}</div>
+    <input
+      v-model="reference"
+      v-on:keyup.enter="handleKeyUp"
+      v-on:click="handleClick"
+      type="text"
+      class="form-control form-control-sm"
+    />
+  </div>
+</template>
+
+<script>
+  import gql from 'graphql-tag';
+  import WIDGETS_NS, { URN } from '@scaife-viewer/scaife-widgets';
+
+  export default {
+    name: 'PassageReferenceWidget',
+    scaifeConfig: {
+      displayName: 'Passage Reference',
+    },
+    watch: {
+      passage: {
+        handler: 'setInputRef',
+        immediate: true,
+      },
+    },
+    computed: {
+      firstPassageUrn() {
+        return this.$store.getters[`${WIDGETS_NS}/firstPassageUrn`];
+      },
+      passage() {
+        return this.$store.getters[`${WIDGETS_NS}/passage`];
+      },
+    },
+    data() {
+      return {
+        reference: '',
+        requested: '',
+        healed: '',
+        human: '',
+      };
+    },
+    methods: {
+      setInputRef() {
+        if (this.passage) {
+          this.reference = this.passage.reference;
+        }
+      },
+      handleKeyUp() {
+        const urn = `${this.passage.version}${this.reference}`;
+        this.$apollo
+          .query({
+            query: gql`
+              query ValidateReference($urn: String!) {
+                passageTextParts(reference: $urn) {
+                  metadata {
+                    healedPassage
+                    humanReference
+                  }
+                }
+              }
+            `,
+            variables: { urn },
+            skip: this.reference === '',
+          })
+          .then(data => {
+            const {
+              healedPassage,
+              humanReference,
+            } = data.data.passageTextParts.metadata;
+            this.human = humanReference;
```

Comment:
As discussed on the PR body, we need the human reference to be reactive to the URN (not just when the user changes the passage via the passage reference widget):

There's also a bug with the human readable display (not being reactive to the changing URN from a user clicking, rather than typing in the passage reference):


https://share.getcloudapp.com/L1uJn21n

Reply:
@paltman: we *do* want the human readable ref on the passage reference widget (https://trello.com/c/S1K7zE4h);

I was suggesting that the display be reactive to the value of `urn` (to handle when the user changes the passage by navigating, not just the reference widget)

Reply:
no problem.  I'll add them back and make them reactive.  I thought you were saying you didn't want them in this PR but instead wanted them in their own PR.